src/Linux/Avalonia.LinuxFramebuffer/Output/DrmBindings.cs

```diff
@@ -144,31 +145,43 @@ public unsafe class DrmCard : IDisposable
         public int Fd { get; private set; }
         public DrmCard(string path = null)
         {
-            if(path == null)
-            {
-                var files = Directory.GetFiles("/dev/dri/");
-
-                foreach(var file in files) 
-                {
-                    var match = Regex.Match(file, "card[0-9]+");
-
-                    if(match.Success)
-                    {
-                        Fd = open(file, 2, 0);
-                        if(Fd != -1) break; 
-                    }    
-                }
+            path = path ?? "/dev/dri/card0";
```

Comment:
I would create
1. private constructor with the Fd as a parameter
2. create a method bool TryOpenCardByPath(string path, out int fd)
3. call TryOpenCardByPath in the public constructor
4. call TryOpenCardByPath in TryCreate and call the private constructor

with that you would not have to fiddle with catching the exception